examples/rag/test_distributed_retriever.py

```diff
@@ -175,50 +214,111 @@ def get_dummy_custom_hf_index_retriever(self, init_retrieval: bool, from_disk: b
+            del dataset
+            retriever = RagRayDistributedRetriever(
+                config,
+                question_encoder_tokenizer=self.get_dpr_tokenizer(),
+                generator_tokenizer=self.get_bart_tokenizer(),
+                retrieval_workers=workers,
+                index=CustomHFIndex.load_from_disk(
+                    vector_size=config.retrieval_vector_size,
+                    dataset_path=config.passages_path,
+                    index_path=config.index_path,
+                ),
+            )
+        else:
+            retriever = RagRayDistributedRetriever(
+                config,
+                question_encoder_tokenizer=self.get_dpr_tokenizer(),
+                generator_tokenizer=self.get_bart_tokenizer(),
+                retrieval_workers=workers,
+                index=CustomHFIndex(config.retrieval_vector_size, dataset),
+            )
+        if init_retrieval:
+            retriever.init_retrieval()
+        return retriever
+
     @require_torch_non_multi_gpu_but_fix_me
-    def test_pytorch_distributed_retriever_retrieve(self):
+    def test_distributed_retriever_retrieve(self):
         n_docs = 1
-        retriever = self.get_dummy_pytorch_distributed_retriever(init_retrieval=True)
         hidden_states = np.array(
             [np.ones(self.retrieval_vector_size), -np.ones(self.retrieval_vector_size)], dtype=np.float32
         )
-        retrieved_doc_embeds, doc_ids, doc_dicts = retriever.retrieve(hidden_states, n_docs=n_docs)
-        self.assertEqual(retrieved_doc_embeds.shape, (2, n_docs, self.retrieval_vector_size))
-        self.assertEqual(len(doc_dicts), 2)
-        self.assertEqual(sorted(doc_dicts[0]), ["embeddings", "id", "text", "title"])
-        self.assertEqual(len(doc_dicts[0]["id"]), n_docs)
-        self.assertEqual(doc_dicts[0]["id"][0], "1")  # max inner product is reached with second doc
-        self.assertEqual(doc_dicts[1]["id"][0], "0")  # max inner product is reached with first doc
-        self.assertListEqual(doc_ids.tolist(), [[1], [0]])
+
+        def test_retriever(retriever):
+            retrieved_doc_embeds, doc_ids, doc_dicts = retriever.retrieve(hidden_states, n_docs=n_docs)
+            self.assertEqual(retrieved_doc_embeds.shape, (2, n_docs, self.retrieval_vector_size))
+            self.assertEqual(len(doc_dicts), 2)
+            self.assertEqual(sorted(doc_dicts[0]), ["embeddings", "id", "text", "title"])
+            self.assertEqual(len(doc_dicts[0]["id"]), n_docs)
+            self.assertEqual(doc_dicts[0]["id"][0], "1")  # max inner product is reached with second doc
+            self.assertEqual(doc_dicts[1]["id"][0], "0")  # max inner product is reached with first doc
+            self.assertListEqual(doc_ids.tolist(), [[1], [0]])
+
+        test_retriever(self.get_dummy_pytorch_distributed_retriever(init_retrieval=True))
+        test_retriever(self.get_dummy_ray_distributed_retriever(init_retrieval=True))
```

Comment:
Could you separate the pytorch test and the ray test in different functions ?
And add a `require_ray` only for the one that uses ray ?

Reply:
Done!